src/dapps/unstoppable-domains/components/UnstoppableDomainBuyOverlay.vue

```diff
@@ -136,9 +199,191 @@ export default {
     }
   },
   data() {
-    return {};
+    return {
+      crypto: true,
+      credit: false,
+      complete: false,
+      paymentError: '',
+      coinbaseError: '',
+      confirmationStep: false,
+      publishableKey: 'pk_live_HAPE6Nv5bfhCJYKe6Nfaaj4P',
+      chargeToken: undefined,
+      udAddress: undefined,
+      emailError: false,
+      loading: false,
+      buyStatus: undefined
+    };
+  },
+  computed: {
+    ...mapState('wallet', ['balance', 'address', 'web3']),
+    convertedEthPrice() {
+      let ethConvertPrice = 0;
+      const domainPrice = this.getDomainPrice();
+      if (domainPrice > 0) {
+        ethConvertPrice = new BigNumber(domainPrice)
+          .dividedBy(this.ethPrice)
+          .toFixed(8);
+      }
+      return ethConvertPrice;
+    },
+    domain() {
+      return this.getDomain();
+    },
+    InsufficientBalance() {
+      const domainPriceInWei = new BigNumber(
+        this.web3.utils.toWei(this.convertedEthPrice, 'ether')
+      );
+      return domainPriceInWei.comparedTo(new BigNumber(this.balance)) === 1;
+    },
+    email() {
+      return this.getEmail();
+    },
+    resellerId() {
+      return this.getResellerId();
+    },
+    order() {
+      return this.getOrder();
+    }
+  },
+  watch: {
+    confirmationStep: async function (newVal) {
+      if (newVal === true) {
+        this.interval = setInterval(
+          async () => await this.updateOrderStatus(),
+          8000
+        );
+      }
+    },
+    order(newVal) {
+      if (newVal.items[0].blockchain.status !== 'PENDING') {
+        clearInterval(this.interval);
+        this.confirmationStep = false;
+        this.close();
+      }
+    }
+  },
+  async mounted() {
+    const coinbaseScript = document.createElement('script');
+    const stripeScript = document.createElement('script');
+    stripeScript.setAttribute('src', 'https://js.stripe.com/v3/');
+    coinbaseScript.setAttribute(
+      'src',
+      'https://commerce.coinbase.com/v1/checkout.js'
+    );
+    document.head.appendChild(coinbaseScript);
+    document.head.appendChild(stripeScript);
+    await this.getEthPrice();
+  },
+  beforeDestroy() {
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+  },
+  methods: {
+    ...mapGetters('unstoppable', [
+      'getDomain',
+      'getEmail',
+      'getDomainPrice',
+      'getDomainName',
+      'getBaseUrl',
+      'getResellerId',
+      'getOrder'
+    ]),
+    ...mapMutations('unstoppable', ['setOrder']),
+    ...mapActions('unstoppable', ['updateOrderStatus']),
+    selectTab(event) {
+      const tabId = event.currentTarget.id;
+      this.crypto = tabId === 'crypto';
+      this.credit = !this.crypto;
+    },
+    async getEthPrice() {
```

Comment:
You can replace this by importing eth value from external store:
`...mapState('external', ['ETHUSDValue']),` and replacing line 223-226 with:
```
ethConvertPrice = new BigNumber(domainPrice)
          .dividedBy(this.ETHUSDValue.value)
          .toFixed(8);
      }
```